src/impls/activity_creator.rs

```diff
@@ -0,0 +1,99 @@
+use crate::interfaces::ActivityCreatorAction;
+use crate::Contract;
+use crate::*;
+use near_contract_standards::non_fungible_token::metadata::TokenMetadata;
+use near_sdk::{env, AccountId};
+
+#[near_bindgen]
+impl ActivityCreatorAction for Contract {
+    #[payable]
+    fn create_activity_token_metadata(&mut self, token_metadata: TokenMetadata) -> ActivityId {
+        let prev_storage = env::storage_usage();
+
+        let activity_id = self.assign_activity_id();
+        let activity_creator = env::predecessor_account_id();
+
+        assert!(
+            self.creator_whitelist.contains(&activity_creator),
+            "{} is not in whitelist.",
+            activity_creator
+        );
+
+        self.activities.insert(&activity_id, &activity_creator);
+
+        let mut creator_activities = self
+            .activities_by_creators
+            .get(&activity_creator)
+            .unwrap_or(vec![]);
+        creator_activities.push(activity_id);
+        self.activities_by_creators
+            .insert(&activity_creator, &creator_activities);
+
+        self.activity_token_metadata
```

Comment:
It's better to check the token metadata is valid before save it.

Reply:
The creators should be responsible for the token metadata, and if they find they set uncorrect token metadata, they can change token metadata by this method: `update_activity_token_metadata`.